### ISSUE TITLE
pre-aggregator sql physical plan optimize 

### ISSUE THREAD

**Issue by zhanghaohit** [4paradigm/OpenMLDB#1207]
**Is your feature request related to a problem? Please describe.**
sub-task of long-time window feature

**Describe the solution you'd like**
add one extra pass
- transform the aggregation node to reduce node (Reduce)
- transform the request union node to request union with aggregation table (RequestUnionAgg)

**Additional context**
long-time window feature

It depends on #1206

**Comment by zhanghaohit** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [long time window - SQL engine](https://trello.com/c/0Ad2TWFJ/4-long-time-window-sql-engine)

**Comment by jingchen2222** [comment]
How do you deal with the merged windows? Check https://github.com/4paradigm/OpenMLDB/blob/release/0.4.0/hybridse/src/plan/planner.cc#L919 for more details about window merge.

```sql
SELECT sum(col1) over w_30d, sum(col2) over w_7d, sum(col3) over w_1d FROM t1
window 
w_30d as (PARTITION BY user ORDER BY std_time ROWS_RANGE BETWEEN 30d PRECEDING AND CURRENT ROW),
w_7d as (PARTITION BY user ORDER BY std_time ROWS_RANGE BETWEEN 7d PRECEDING AND CURRENT ROW),
w_1d as (PARTITION BY user ORDER BY std_time ROWS_RANGE BETWEEN 1d PRECEDING AND CURRENT ROW)
```
The SQL script above aggregates features over 3 windows. These windows are overlapped with each other. So basically, we can regard them as a sub-window of the same big window (30 days window) so that the logical plan will only generate one WindowProjectPlan.

When we try to perform a long-window optimization on these windows aggregation, we definitely should keep the original windows and generate three WindowProjectPlan so that we can transform RequestUnionAgg/Reduce for each long window.

In summary, we might need to disable the window auto-merged when we are applying long-window optimization. 
@zhanghaohit

**Comment by zhanghaohit** [comment]
> How do you deal with the merged windows? Check https://github.com/4paradigm/OpenMLDB/blob/release/0.4.0/hybridse/src/plan/planner.cc#L919 for more details about window merge.
> 
> ```sql
> SELECT sum(col1) over w_30d, sum(col2) over w_7d, sum(col3) over w_1d FROM t1
> window 
> w_30d as (PARTITION BY user ORDER BY std_time ROWS_RANGE BETWEEN 30d PRECEDING AND CURRENT ROW),
> w_7d as (PARTITION BY user ORDER BY std_time ROWS_RANGE BETWEEN 7d PRECEDING AND CURRENT ROW),
> w_1d as (PARTITION BY user ORDER BY std_time ROWS_RANGE BETWEEN 1d PRECEDING AND CURRENT ROW)
> ```
> 
> The SQL script above aggregates features over 3 windows. These windows are overlapped with each other. So basically, we can regard them as a sub-window of the same big window (30 days window) so that the logical plan will only generate one WindowProjectPlan.
> 
> When we try to perform a long-window optimization on these windows aggregation, we definitely should keep the original windows and generate three WindowProjectPlan so that we can transform RequestUnionAgg/Reduce for each long window.
> 
> In summary, we might need to disable the window auto-merged when we are applying long-window optimization. @zhanghaohit

Sorry for missing your comments. 

Yes, you are right. Currently I do a `Split` pass to split the `aggregation node with multiple aggregations` to multiple aggregation nodes to achieve this.
